### PR TITLE
Summary text improvement

### PR DESCRIPTION
**Pull request by furkanmtorun**
Hi Max,

In this PR, I fixed the few issues addressed in #4 .
- Added "," comma between parameters
- Added xgboost version to the text
- Fixed the imputation mistake

Screenshot 2023-04-01 at 17 32 07

### CODE REVIEW

File: omiclearn/utils/ui_components.py [hunk 1]
```diff
@@ -896,32 +897,32 @@ def generate_summary_text(state, report):
...
 
-    # Missing value impt.
-    if state.missing_value != "None":
-        text += "To impute missing values, a {}-imputation strategy is used. ".format(
-            state.missing_value
-        )
+    # Missing value imptutation
+    if state.n_missing > 0:
+        if state.missing_value != "None":
+            text += f"To impute missing values, a {state.missing_value}-imputation strategy is used. "
```
Comment (straussmaximilian) [inline omiclearn/utils/ui_components.py]: We used past tense before, so this should also be past tense (was used)

File: omiclearn/utils/ui_components.py [hunk 2]
```diff
@@ -896,32 +897,32 @@ def generate_summary_text(state, report):
...
+        else:
+            text += "Even though dataset contained missing values; no imputation was performed. "
```
Comment (straussmaximilian) [inline omiclearn/utils/ui_components.py]: "Even though" puts things into contrast and one could understand this as one should have chosen missing value impoutation.

Suggestion to be more neutral sounding: "Missing values were not imputed."

File: omiclearn/utils/ui_components.py [hunk 3]
```diff
@@ -896,32 +897,32 @@ def generate_summary_text(state, report):
...
         text += "The dataset contained no missing values; hence no imputation was performed. "
 
     # Features
     if state.feature_method == "None":
         text += "No feature selection algorithm was applied. "
     elif state.feature_method == "ExtraTrees":
-        text += "Features were selected using a {} (n_trees={}) strategy with the maximum number of {} features. ".format(
-            state.feature_method, state.n_trees, state.max_features
-        )
+        text += f"Features were selected using a {state.feature_method} (n_trees={state.n_trees}) strategy with the maximum number of {state.max_features} features. "
```
Comment (straussmaximilian) [inline omiclearn/utils/ui_components.py]: "... with **a** maximum number of"

File: omiclearn/utils/ui_components.py [hunk 4]
```diff
@@ -896,32 +897,32 @@ def generate_summary_text(state, report):
...
     else:
-        text += "Features were selected using a {} strategy with the maximum number of {} features. ".format(
-            state.feature_method, state.max_features
-        )
+        text += f"Features were selected using a {state.feature_method} strategy with the maximum number of {state.max_features} features. "
```
Comment (straussmaximilian) [inline omiclearn/utils/ui_components.py]: "... with **a** maximum number of"

File: omiclearn/utils/ui_components.py [hunk 5]
```diff
@@ -896,32 +897,32 @@ def generate_summary_text(state, report):
...
     text += "During training, normalization and feature selection was individually performed using the data of each split. "
 
     # Classification
     params = [f"{k} = {v}" for k, v in state.classifier_params.items()]
-    text += f"For classification, we used a {state.classifier}-Classifier ({' '.join(params)}). "
+    if state.classifier == "XGBoost":
+        text += f"For classification, we used a {state.classifier}-Classifier (version: {report['xgboost_version']}, {', '.join(params)}). "
```
Comment (straussmaximilian) [inline omiclearn/utils/ui_components.py]: Before that we used passive voice and no pronouns ("we"), I would keept his and be consistent:

"... we used an XGBoost classifier"

**Review by straussmaximilian — CHANGES_REQUESTED** [review]
Would suggest to change as suggested.

File: omiclearn/utils/ui_components.py (hunk 1 above)
Comment (furkanmtorun) [reply omiclearn/utils/ui_components.py]: Now, changed.

File: omiclearn/utils/ui_components.py (hunk 2 above)
Comment (furkanmtorun) [reply omiclearn/utils/ui_components.py]: Really great point! Thank you very much. Changed now

File: omiclearn/utils/ui_components.py (hunk 3 above)
Comment (furkanmtorun) [reply omiclearn/utils/ui_components.py]: Typical grammar mistake for me 😸 Fixed

File: omiclearn/utils/ui_components.py (hunk 4 above)
Comment (furkanmtorun) [reply omiclearn/utils/ui_components.py]: fixed :)

File: omiclearn/utils/ui_components.py (hunk 5 above)
Comment (furkanmtorun) [reply omiclearn/utils/ui_components.py]: Good catch! Fixed now

```python
if state.classifier == "XGBoost":
        text += f"For classification, {state.classifier}-Classifier (version: {report['xgboost_version']}, {', '.join(params)}) was used. "
    else:
        text += f"For classification, {state.classifier}-Classifier ({', '.join(params)}) was used. "
```

**Review by straussmaximilian — APPROVED** [review]
LGTM

**Pull request merged**